proposals/hasCrossSiteAncestor.md

```diff
@@ -0,0 +1,154 @@
+To interact with partitioned cookies containing a cross-site ancestor chain bit correctly, extensions will need to have the ability to to specify a value (`hasCrossSiteAncestor`) that corresponds to the value of the cross-site ancestor chain bit in partitioned cookies.
+
+#### Use Cases
+
+#### Cookie Manager:
+Let's say a cookie manager extension (with host permissions) is used by users to get/set/remove their cookies. As browsers include the cross-site ancestor chain bit in their implementation of partitioned cookies, the extension will need the ability to use the `hasCrossSiteAncestor` parameter to give full insight into the existing cookies and allow the user to set new cookies that include the cross-site ancestor bit correctly.
+
+#### Password Manager:
+Let’s say a password manager extension (with host permissions) is used by users to access their login information by setting a cookie that stores their usernames and passwords in an encrypted partitioned cookie. To protect their users against clickjacking, the extension adds a setting that prevents their cookies from being accessed, by default, in embeds that have cross site ancestors without triggering a user prompt. If permission is given through the prompt, the extension sets a cookie with a `hasCrossSiteAncestor` value of true. Upon subsequent visits, the extension checks the cookie store for the presence of a cookie with a `hasCrossSiteAncestor` value of true to determine whether the prompt needs to be rendered.
+
+To allow for this protection and UX flow to work, the extension would need to have the ability to set/get cookies with specific `hasCrossSiteAncestor` values. 
+
+### Known Consumers
+All extensions that access and/or modify cookies with awareness of partitioned cookies, through the use of the `partitionKey` property in the `cookies` extension API.
+
+## Specification
+
+### Schema
+
+Adds a new boolean property `hasCrossSiteAncestor` to the `partitionKey` property of the `cookies.Cookie` type.
+
+Adds a new optional boolean property `hasCrossSiteAncestor` to the `partitionKey` property of the details object. Which is the parameter used in the methods `cookies.get()`, `cookies.set()`, `cookies.remove()` and `cookies.getAll()`.
+
+```
+// Cookie details object with partitionKey containing hasCrossSiteAncestor value of true.
+"details": {
+  "partitionKey": {
+    "topLevelSite": "https://example.com",
+    "hasCrossSiteAncestor": true
+  }
+}
+```
+#### Cookies.GetPartitionKey()
+Adds a new API, `cookies.getPartitionKey()` which retrives a valid `partitionKey` for the indicated frame.
+
+This is an asyncronus function that returns a promise.
+
+##### Syntax
+```
+let key = cookies.getPartitionKey(
+    details // object
+)
+```
+##### Parameters
+`details` object. Information about the frame to retrive information about.
+>  `tabId`
+integer. The ID of the tab in which the frame is.
+
+> `frameId`
+integer. The ID of the frame in the given tab.
+
+##### Return value
```

Comment:
Could you specify:
- Does this method always return a partition key? If not, when?
- Are there any (origin) permission requirements? Cautiously we could opt for requiring host permissions for the target frame, because that permission is also required to execute a script there, which in turn could query the top level origin via `location.ancestorOrigins`. I am also willing to consider not requiring origins if we deem the return value to not be too sensitive. After all, extensions can already observe partitioned cookies for non-document subresources in frames through the cookies API.
- Are there error conditions? E.g. frame does not exist, permission denied.

Reply:
>Does this method always return a partition key? If not, when?

It should always return a partitionKey unless an error occurs.

>Are there any (origin) permission requirements?

In the chromium implementation, `cookies.get()`, `cookies.set()` and `cookies.remove()` all require host permissions so I think it makes sense to be consistent here as well. 

> Are there error conditions? 

Both the frame does not exist and permission denied would be an error. 
Do you have any thoughts on if a mismatch between the documentId and any tabId/frameId combination should be treated as an error?

Reply:
This thread was marked as resolved, but the updated PR does not mention anything about permission requirements or errors. Did you forget to push a commit with the updated text?

Reply:
Yes, pushed the commit